sdk/servicebus/azure-servicebus/azure/servicebus/_transport/_pyamqp_transport.py

```diff
@@ -699,35 +699,42 @@ def settle_message_via_receiver_link(
         dead_letter_error_description: Optional[str] = None,
     ) -> None:
         # pylint: disable=protected-access
-        if settle_operation == MESSAGE_COMPLETE:
```

Comment:
is there a repro that is consistent, so that we can add tests for this?

Reply:
there is a repro that I have used for debugging, but will see if I can repro using sleep in a live test. Its a long running one because it requires 0 activity for 5 mins to get to this point.

Reply:
cant make a live test out of this using sleep. I need to block the entire process to stop and prevent any keep alives etc to run in the background